src/classes/MyHandler/MyHandler.ts

```diff
@@ -1243,20 +1247,50 @@ export default class MyHandler extends Handler {
         const uncraftAll = this.bot.uncraftWeapons;
 
         const itemsDiff = offer.getDiff();
-        //?
+        /* this loop goes through the following
+        buying = false; which = 'our';   intent = 'sell';
+        buying = true;  which = 'their'; intent = 'buy';
+         */
         for (let i = 0; i < states.length; i++) {
             const buying = states[i];
             const which = buying ? 'their' : 'our';
             const intentString = buying ? 'buy' : 'sell';
 
+            // TODO: Go through all assetids and check if the item is being sold for a specific price prior to processing the whole sku
+            if ('our' === which) {
+                for (const sku in items['our']) {
+                    if (!Object.prototype.hasOwnProperty.call(items['our'], sku)) {
+                        continue;
+                    }
+                    items['our'][sku] = items['our'][sku].filter(item => {
+                        const match: Entry | null = this.bot.pricelist.getPrice(item.id);
+                        // if we have a price we want to filter this item out from the generic sku items
+                        const hasAssetPrice = null !== match;
+                        if (hasAssetPrice) {
+                            // Add value of items
+                            exchange[which].value += match[intentString].toValue(keyPrice.metal);
+                            exchange[which].keys += match[intentString].keys;
+                            exchange[which].scrap += Currencies.toScrap(match[intentString].metal);
+                            itemPrices[item.id] = {
+                                buy: match.buy,
+                                sell: match.sell
+                            };
+                        }
+                        return !hasAssetPrice;
+                    });
+                }
+            }
```

Comment:
And then use that `isAssetidPriceEnabled` together with `'our' === which`

```js
if (isAssetidPriceEnabled && 'our' === which) {
    // ...
}
```

Reply:
this is all removed